### PR TITLE
Cb lang

### PR DESCRIPTION
**Pull request by maehr**
# Pull request

## Proposed changes

- Live https://maehr.github.io/collectionbuilder-csv/

- See https://github.com/CollectionBuilder/cb-translate/issues/4
- See https://github.com/CollectionBuilder/cb-translate/issues/5
- See https://github.com/CollectionBuilder/cb-translate/pull/10
- See https://github.com/CollectionBuilder/cb-translate/pull/7

## Types of changes

- [x] New feature (non-breaking change which adds functionality).
- [ ] Enhancement (non-breaking change which enhances functionality)
- [ ] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Summary by CodeRabbit

- **New Features**
	- Enhanced multilingual support across the site, dynamically fetching translated text for various elements based on the site's language setting.
	- Improved user experience by updating error messages and download link messages in the video player component.
	- Updated button labels, modal titles, and descriptive text to be language-dependent for a localized user experience.

- **Bug Fixes**
	- Fixed static error messages in the video player component to display dynamic, language-dependent messages for better user interaction.

- **Documentation**
	- Updated documentation to include details about the new multilingual support and dynamic content updates.

### CODE REVIEW

**Comment by maehr** [comment]
@koilebeit for bug hunting I set up https://maehr.github.io/collectionbuilder-csv/
There are many 😅

File: _includes/data-download-modal.html [hunk 1]
```diff
@@ -6,81 +6,82 @@
...
                     </div>
                 </div>
                 {%- endif -%}
                 {% if stubs contains "map" %}
                 <div class="card my-3">
                     <div class="card-body">
                         <h2 class="card-title h4">GeoJSON</h2>
-                        <p class="card-text">Metadata for all collection items that have geographic coordinates in <a href="https://en.wikipedia.org/wiki/GeoJSON" target="_blank" rel="noopener">GeoJSON</a> format, useful for further exploration and analysis of this collection through a geographical lense.</p>
+                        <p class="card-text">{{ site.data.translations['_includes']['data-download-modal.html']['geojson-description'][site.lang] | default: "Metadata for all collection items that have geographic coordinates in <a href='https://en.wikipedia.org/wiki/GeoJSON' target='_blank' rel='noopener'>GeoJSON</a> format, useful for further exploration and analysis of this collection through a geographical lense." }}</p>
```
[the remark this answers is not in the document]
Comment (maehr) [reply _includes/data-download-modal.html:66]: Yes. That's right. The translation needs to be added to `data/translations.yml` as well.

File: _includes/cb/credits.html [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
+                {{ site.data.translations['_includes']['cb']['credits.html']['content'][site.lang] | default: "<p>This digital collection is built with <a href='https://collectionbuilder.github.io/' target='_blank' rel='noopener'>CollectionBuilder</a>, an open source framework for creating digital collection and exhibit websites that is developed by faculty librarians at the University of Idaho Library following the <a href='https://lib-static.github.io' target='_blank' rel='noopener'>Lib-Static</a> methodology.</p><p>Using the <a href='https://github.com/CollectionBuilder/collectionbuilder-csv' target='_blank' rel='noopener'>CollectionBuilder-CSV</a> template and the static website generator <a href='https://jekyllrb.com/' target='_blank' rel='noopener'>Jekyll</a>, this project creates an engaging interface to explore driven by metadata.</p>" }}
```
Comment (maehr) [inline _includes/cb/credits.html:6]: Add newline

File: _includes/data-download-modal.html [hunk 3]
```diff
@@ -6,81 +6,82 @@
...
                 
                 {% if site.data.theme.metadata-export-fields %}
                 <div class="card my-3">
                     <div class="card-body">
-                        <h2 class="card-title h4">Complete Metadata</h2>
-                        <p class="card-text">All metadata fields for all collection items, available as a CSV spreadsheet (usable in Excel, Google Sheets, and similar programs) or JSON file (often used with web applications).</p>
-                        <a href="{{ '/assets/data/metadata.csv' | relative_url }}" class="btn btn-outline-dark" target="_blank">Metadata CSV</a>
-                        <a href="{{ '/assets/data/metadata.json' | relative_url }}" class="btn btn-outline-dark" target="_blank">Metadata JSON</a>
+                        <h2 class="card-title h4">{{ site.data.translations['_includes']['data-download-modal.html']['complete-metadata'][site.lang] | default: 'Complete Metadata' }}</h2>
+                        <p class="card-text">
```
Comment (maehr) [inline _includes/data-download-modal.html:25]: Remove newline

File: _includes/data-download-modal.html [hunk 4]
```diff
@@ -6,81 +6,82 @@
...
+                            {{ site.data.translations['_includes']['data-download-modal.html']['metadata-description'][site.lang] | default: 'All metadata fields for all collection items, available as a CSV spreadsheet (usable in Excel, Google Sheets, and similar programs) or JSON file (often used with web applications).' }}
```
Comment (maehr) [inline _includes/data-download-modal.html:26]: Remove newline

File: _includes/data-download-modal.html [hunk 5]
```diff
@@ -6,81 +6,82 @@
...
                     </div>
                 </div>
                 {%- endif -%}
                 {% if stubs contains "timeline" %}
                 <div class="card my-3">
                     <div class="card-body">
-                        <h2 class="card-title h4">Timeline</h2>
-                        <p class="card-text">A time-focused JSON data export designed for use with <a href="https://timeline.knightlab.com/" target="_blank" rel="noopener">TimelineJS</a>.</p>
+                        <h2 class="card-title h4">{{ site.data.translations['_includes']['data-download-modal.html']['timeline'][site.lang] | default: 'Timeline' }}</h2>
+                        <p class="card-text">{{ site.data.translations['_includes']['data-download-modal.html']['timeline-description'][site.lang] | default: "Time-based metadata formatted for <a href='https://timeline.knightlab.com/' target='_blank' rel='noopener'>TimelineJS</a> or other applications, useful for further exploration and analysis of this collection's temporal aspects." }}</p>
```
Comment (maehr) [inline _includes/data-download-modal.html:75]: Eng text changed

File: _includes/data-download-modal.html [hunk 6]
```diff
@@ -6,81 +6,82 @@
...
                         <a href="{{ '/assets/data/timelinejs.json' | relative_url }}" class="btn btn-outline-dark" target="_blank">TimelineJS JSON</a>
                     </div>
                 </div>
                 {%- endif -%}
                 <div class="card my-3">
                     <div class="card-body">
-                        <h2 class="card-title h4">Website Source Code</h2>
-                        <p class="card-text">GitHub repository containing source code for this project built with <a href="https://github.com/CollectionBuilder/collectionbuilder-csv" target="_blank" rel="noopener">CollectionBuilder-CSV</a>.</p>
-                        <a href="{{ site.source-code | default: 'https://github.com/CollectionBuilder/collectionbuilder-csv' }}" class="btn btn-outline-dark" target="_blank" rel="noopener">Source Code</a>
+                        <h2 class="card-title h4">{{ site.data.translations['_includes']['data-download-modal.html']['source-code'][site.lang] | default: 'Source Code' }}</h2>
```
Comment (maehr) [inline _includes/data-download-modal.html:82]: Text changed

File: _includes/feature/nav-menu.html [hunk 7]
```diff
@@ -13,5 +13,5 @@
 {%- endcomment -%}
 {% assign sections = include.sections | split: ';'%}
 <p class="h6 shadow-sm p-3 about-nav sticky-top bg-white">
-    Contents: {% for section in sections %}<a class="mx-2" href="#{{ section | slugify }}">{{ section }}</a> {% unless forloop.last %}| {% endunless %}{% endfor %}{% if page.credits == true %}| <a class="mx-2" href="#technical">Tech</a>{% endif %}
+    {{ site.data.translations['_includes']['feature']['nav-menu.html']['contents'][site.lang] | default: "Contents" }} {% for section in sections %}<a class="mx-2" href="#{{ section | slugify }}">{{ section }}</a> {% unless forloop.last %}| {% endunless %}{% endfor %}{% if page.credits == true %}| <a class="mx-2" href="#technical">{{ site.data.translations['_includes']['feature']['nav-menu.html']['tech'][site.lang] | default: "Tech" }}</a>{% endif %}
```
Comment (maehr) [inline _includes/feature/nav-menu.html:16]: "Contents:"

File: _includes/feature/pdf.html [hunk 8]
```diff
@@ -35,7 +35,8 @@
     <figure class="figure border rounded p-1 feature-w-{{ include.width | default: '100' }}">
         <div class="{% unless include.height %}ratio ratio-{{ include.ratio | default: '1x1' }}{% endunless %}">
             <object aria-label="pdf embed of {{ pdf_caption | escape }}" {% if include.height %}width="100%" height="{{ include.height }}"{% endif %} data="{{ src }}">
-                <p>The PDF is not rendering in your browser. Please <a href="{{ src }}">download the PDF</a> to view.</p>
+                <p>{{ site.data.translations['_includes']['feature']['pdf.html']['error'][site.lang] 
+                     | default: "Your browser does not support the PDF viewer." }} <a href="{{ src }}">{{ site.data.translations['_includes']['feature']['pdf.html']['download'][site.lang] | default: "Download PDF" }}</a></p>
```
Comment (maehr) [inline _includes/feature/pdf.html:39]: Newline plustext changed

File: _includes/feature/video.html [hunk 9]
```diff
@@ -49,7 +49,7 @@
             {% else %}
             <video width="320" height="240" controls>
                 <source src="{{ video_src }}">
-                Your browser does not support the video tag.
+                {{ site.data.translations['_includes']['feature']['video.html']['error'][site.lang] | default: "Your browser does not support the video element." }}
```
Comment (maehr) [inline _includes/feature/video.html:52]: Text changed

File: _includes/index/objects.html [hunk 10]
```diff
@@ -2,35 +2,35 @@
...
     <p class="card-text">
       {% for t in types %}
       {%- assign count = templates | where_exp: 'item','item contains t' | size -%}
       {% if count > 0 %}
       {{ count }}      
       {% case t %}
       {% when 'image' %} 
-      <a class="text-dark" href="{{ 'image' | prepend: '/browse.html#' | relative_url }}">Image{% if count > 1 %}s{% endif %} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-image"/></svg></a><br>
+      <a class="text-dark" href="{{ 'image' | prepend: '/browse.html#' | relative_url }}">{{ site.data.translations['_includes']['index']['objects.html']['image'][site.lang] | default: "Image(s)" }} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-image"/></svg></a><br>
       {% when 'audio' %}
```
Comment (maehr) [inline _includes/index/objects.html:14]: Text changed

File: _includes/index/objects.html [hunk 11]
```diff
@@ -2,35 +2,35 @@
...
-      <a class="text-dark" href="{{ 'audio' | prepend: '/browse.html#' | relative_url }}">Audio File{% if count > 1 %}s{% endif %} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-audio"/></svg></a><br>
+      <a class="text-dark" href="{{ 'audio' | prepend: '/browse.html#' | relative_url }}">{{ site.data.translations['_includes']['index']['objects.html']['audio'][site.lang] | default: "Audio(s)" }} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-audio"/></svg></a><br>
```
Comment (maehr) [inline _includes/index/objects.html:15]: Text changed

File: _includes/index/objects.html [hunk 12]
```diff
@@ -2,35 +2,35 @@
...
       {% when 'video' %}
-      <a class="text-dark" href="{{ 'video' | prepend: '/browse.html#' | relative_url }}">Video{% if count > 1 %}s{% endif %} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-video"/></svg></a><br>
+      <a class="text-dark" href="{{ 'video' | prepend: '/browse.html#' | relative_url }}">{{ site.data.translations['_includes']['index']['objects.html']['video'][site.lang] | default: "Video(s)" }}  <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-video"/></svg></a><br>
```
Comment (maehr) [inline _includes/index/objects.html:17]: Text changed

File: _includes/index/objects.html [hunk 13]
```diff
@@ -2,35 +2,35 @@
...
       {% when 'pdf' %}
-      <a class="text-dark" href="{{ 'pdf' | prepend: '/browse.html#' | relative_url }}">PDF{% if count > 1 %}s{% endif %} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-pdf"/></svg></a><br>
+      <a class="text-dark" href="{{ 'pdf' | prepend: '/browse.html#' | relative_url }}">{{ site.data.translations['_includes']['index']['objects.html']['pdf'][site.lang] | default: "PDF File(s)" }}  <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-pdf"/></svg></a><br>
```
Comment (maehr) [inline _includes/index/objects.html:19]: Text changed

File: _includes/index/objects.html [hunk 14]
```diff
@@ -2,35 +2,35 @@
...
       {% when 'panorama' %}
-      <a class="text-dark" href="{{ 'panorama' | prepend: '/browse.html#' | relative_url }}">Panorama{% if count > 1 %}s{% endif %} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-panorama"/></svg></a><br>
+      <a class="text-dark" href="{{ 'panorama' | prepend: '/browse.html#' | relative_url }}">{{ site.data.translations['_includes']['index']['objects.html']['panorama'][site.lang] | default: "Panorama(s)" }}  <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-panorama"/></svg></a><br>
```
Comment (maehr) [inline _includes/index/objects.html:21]: Text changed

File: _includes/index/objects.html [hunk 15]
```diff
@@ -2,35 +2,35 @@
...
+      <a class="text-dark" href="{{ 'record' | prepend: '/browse.html#' | relative_url }}">{{ site.data.translations['_includes']['index']['objects.html']['record'][site.lang] | default: "Record(s)" }}  <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-record"/></svg></a><br>
       {% when 'compound_object' %}
-      <a class="text-dark" href="{{ 'compound%20object' | prepend: '/browse.html#' | relative_url }}">Compound Object{% if count > 1 %}s{% endif %} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-compound-object"/></svg></a><br>
+      <a class="text-dark" href="{{ 'compound%20object' | prepend: '/browse.html#' | relative_url }}">{{ site.data.translations['_includes']['index']['objects.html']['compound_object'][site.lang] | default: "Compound Object(s)" }}  <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-compound-object"/></svg></a><br>
```
Comment (maehr) [inline _includes/index/objects.html:25]: Text changed

File: _includes/index/objects.html [hunk 16]
```diff
@@ -2,35 +2,35 @@
...
       {% when 'multiple' %}
-      <a class="text-dark" href="{{ 'multiple' | prepend: '/browse.html#' | relative_url }}">Multiple{% if count > 1 %}s{% endif %} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-multiple"/></svg></a><br>
+      <a class="text-dark" href="{{ 'multiple' | prepend: '/browse.html#' | relative_url }}">{{ site.data.translations['_includes']['index']['objects.html']['multiple'][site.lang] | default: "Multiple(s)" }}  <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-multiple"/></svg></a><br>
```
Comment (maehr) [inline _includes/index/objects.html:27]: Text changed

File: _includes/index/objects.html [hunk 17]
```diff
@@ -2,35 +2,35 @@
...
       {% else %}
-      <a class="text-dark" href="{{ 'file' | prepend: '/browse.html#' | relative_url }}">File{% if count > 1 %}s{% endif %} <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-default"/></svg></a><br>
+      <a class="text-dark" href="{{ 'file' | prepend: '/browse.html#' | relative_url }}">{{ site.data.translations['_includes']['index']['objects.html']['file'][site.lang] | default: "File(s)" }}  <svg class="bi icon-sprite text-body" fill="currentColor" aria-hidden="true"><use xlink:href="{{ '/assets/lib/cb-icons.svg' | relative_url }}#icon-default"/></svg></a><br>
```
Comment (maehr) [inline _includes/index/objects.html:29]: Text changed

**Comment by koilebeit** [comment]
Thanks for the huge amount of work. I think there are still some translations missing, especially for buttons like the "view item" button on the carousel on the start page, and the search button on the search page.

Should the markdown pages in the pages folder also be translated? But this would not work via the translations.yml?

**Comment by maehr** [comment]
> Thanks for the huge amount of work. I think there are still some translations missing, especially for buttons like the "view item" button on the carousel on the start page, and the search button on the search page.
> 
> Should the markdown pages in the pages folder also be translated? But this would not work via the translations.yml?

The markdown files will not be translated. Can you point to the code with the missing translations?

**Comment by maehr** [comment]
> > > Thanks for the huge amount of work. I think there are still some translations missing, especially for buttons like the "view item" button on the carousel on the start page, and the search button on the search page.
> > > Should the markdown pages in the pages folder also be translated? But this would not work via the translations.yml?
> > 
> > 
> > The markdown files will not be translated. Can you point to the code with the missing translations?
> 
> https://github.com/Stadt-Geschichte-Basel/forschung.stadtgeschichtebasel.ch/blob/5c1864d9bc66f833057853c4237cd6c40c988cf3/_layouts/timeline.html#L26-L28
> 
> https://github.com/Stadt-Geschichte-Basel/forschung.stadtgeschichtebasel.ch/blob/5c1864d9bc66f833057853c4237cd6c40c988cf3/_layouts/timeline.html#L38
> 
> https://github.com/Stadt-Geschichte-Basel/forschung.stadtgeschichtebasel.ch/blob/5c1864d9bc66f833057853c4237cd6c40c988cf3/_layouts/search.html#L36C1-L38C22
> 
> https://github.com/Stadt-Geschichte-Basel/forschung.stadtgeschichtebasel.ch/blob/5c1864d9bc66f833057853c4237cd6c40c988cf3/_includes/index/carousel.html#L23

I fixed all. If you could use the inline comment feature next time https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request#starting-a-review GitHub can automate even more steps of the review.

**Comment by maehr** [comment]
@koilebeit Ready for review. I fixed some more obvious bugs ;)

File: _layouts/search.html [hunk 18]
```diff
@@ -43,6 +43,6 @@ <h5 class="modal-title" id="modalLabel">Lunr Search Options</h5>
 </div>
```
Comment (koilebeit) [inline _layouts/search.html:43]: Here are the search bar and the button not translated.

Also the display of how many results the search returns: 
image
But this comes probably directly from lunr.js?

File: _layouts/timeline.html [hunk 19]
```diff
@@ -24,7 +25,7 @@
 {%- if navYears -%}
 <div class="dropdown float-end" id="year-nav">
     <button class="btn btn-primary dropdown-toggle" type="button" id="yearButton" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-        Year
+        years
```
Comment (koilebeit) [inline _layouts/timeline.html:28]:
```suggestion
        {{ year }}
```
?
but is it good idea to translate variables within liquid code?

**Review by koilebeit — CHANGES_REQUESTED** [review]
It seems that the translation in the search bar in the navigation bar is not working

**Comment by maehr** [comment]
> It seems that the translation in the search bar in the navigation bar is not working: image

Fixed it

Screenshot 2024-02-28 at 11 20 36

File: _layouts/search.html (hunk 18 above)
Comment (maehr) [reply _layouts/search.html:43]: fixed it as well as "Enter your search term..." and Search button.

**Review by koilebeit — APPROVED** [review]

**Pull request merged**